public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -611,6 +614,17 @@ export class LokiDatasource
     return escapedValues.join('|');
   }
 
+  analyzeQuery(query: LokiQuery, options: AnalyzeQueryOptions): boolean {
+    let expression = query.expr ?? '';
+    switch (options.check) {
+      case 'HAS_FILTER': {
+        return queryHasFilter(expression, options.attributes.key, '=', options.attributes.value);
+      }
+      default:
+        return false;
+    }
+  }
+
   modifyQuery(query: LokiQuery, action: QueryFixAction): LokiQuery {
```

Comment:
Can we here also keep original modifyQuery and add a new method for toggling label filters?

Reply:
Done ⛵